desktop/libs/notebook/src/notebook/connectors/base_tests.py

```diff
@@ -114,3 +113,100 @@ def check_status_side_effect(request, operation_id):
   else:
     return {'status': 0, 'query_status': {'status': 'available'}}
+
+
+@pytest.mark.django_db
+class TestConnectorApiCompatibility(object):
+  """
+  Test API compatibility across all connectors to prevent signature mismatches.
+  This ensures that when new parameters are added to the base API, all connectors
+  remain compatible and don't break due to signature differences.
+  """
+
+  def setup_method(self):
+    self.client = make_logged_in_client(username="test_connector_compatibility", groupname="default", recreate=True, is_superuser=False)
+    self.user = User.objects.get(username="test_connector_compatibility")
+
+  def teardown_method(self):
+    User.objects.filter(username="test_connector_compatibility").delete()
+
+  def test_base_api_method_signatures(self):
+    """
+    Test that the base Api class has the expected method signatures that all connectors should follow.
+    """
+    from notebook.connectors.base import Api
+
+    # Check base get_sample_data signature
+    base_method = getattr(Api, 'get_sample_data', None)
+    assert base_method is not None, "Base Api class missing get_sample_data method"
+
+    sig = inspect.signature(base_method)
+    expected_params = {'self', 'snippet', 'database', 'table', 'column', 'nested', 'is_async', 'operation'}
+    actual_params = set(sig.parameters.keys())
+
+    assert expected_params == actual_params, f"Base Api method signature changed. Expected: {expected_params}, Got: {actual_params}"
+
+    # Verify nested parameter has default None
+    nested_param = sig.parameters.get('nested')
+    assert nested_param is not None, "nested parameter missing from base Api"
+    assert nested_param.default is None, f"nested parameter should default to None, got: {nested_param.default}"
+
+  def test_source_code_signature_compatibility(self):
+    """
+    Test connector method signatures by parsing source code directly.
+    This is the most reliable way to check signatures, avoiding decorator interference.
+    """
+    # Define connectors and their file paths
+    connector_files = [
+      ('SqlAlchemy', 'desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py'),
+      ('Spark', 'desktop/libs/notebook/src/notebook/connectors/spark_shell.py'),
+      ('HiveServer2', 'desktop/libs/notebook/src/notebook/connectors/hiveserver2.py'),
+      ('Flink', 'desktop/libs/notebook/src/notebook/connectors/flink_sql.py'),
+      ('JDBC', 'desktop/libs/notebook/src/notebook/connectors/jdbc.py'),
+      ('RDBMS', 'desktop/libs/notebook/src/notebook/connectors/rdbms.py'),
+      ('Solr', 'desktop/libs/notebook/src/notebook/connectors/solr.py'),
+      ('KSQL', 'desktop/libs/notebook/src/notebook/connectors/ksql.py'),
+      ('SQLFlow', 'desktop/libs/notebook/src/notebook/connectors/sqlflow.py'),
+      ('Trino', 'desktop/libs/notebook/src/notebook/connectors/trino.py'),
+      ('HiveMetastore', 'desktop/libs/notebook/src/notebook/connectors/hive_metastore.py'),
```

Comment:
Possible to not hardcode filepaths?

Reply:
I checked, and it's not possible to retrieve the files without specifying the correct path to the function. Otherwise, we would need to search the entire Hue directory to locate the connector path and even then, we’d still need to provide the exact file name. In my opinion, providing the full path is the better approach.